samcli/lib/build/app_builder.py

```diff
@@ -383,6 +403,10 @@ def _build_function(  # pylint: disable=R1710
             Path to the location where built artifacts are available
         """
         if packagetype == IMAGE:
+            if not metadata:
+                raise MissingMetadataForImageFunctionException(
+                    f"{function_name} has Image package type but Metadata is missing."
+                )
```

Comment:
Why does this have to be in this PR? Would make reviewing easier if there was no logic changes.